CorsixTH/Lua/world.lua

```diff
@@ -511,7 +534,15 @@ function World:spawnPatient(hospital)
   end
 
   -- Construct patient.
-  local spawn_point = self.spawn_points[math.random(1, #self.spawn_points)]
+  local desks = hospital:findReceptionDesks()
+  local spawns = self.spawn_points
+  local spawn_point = spawns[math.random(1, #spawns)]
+
+  -- Let the first patient to the player use the smallest path to a reception desk
+  if hospital.num_visitors == 0 and class.is(hospital, PlayerHospital) then
```

Comment:
Condition is becoming more complicated. It looks like a question that you should ask the hospital. By doing that you can handle the class check in a much nicer way too.

Reply:
It's an odd thing to think of how to make the function useful. But yeah we should really be asking the hospital for this so I'll think on it.